opm/autodiff/GeoProps.hpp

```diff
@@ -150,37 +143,22 @@ namespace Opm
             std::vector<double> mult;
             multiplyHalfIntersections_(grid, eclState, ntg, htrans, mult);
 
-            // Handle NNCs
-            if (eclState) {
-                nnc_ = eclState->getNNC();
-            }
-
-            // opmfil is hardcoded to be true. i.e the pinch processor is never used
-            if (!opmfil && eclgrid->isPinchActive()) {
-                const double minpv = eclgrid->getMinpvValue();
-                const double thickness = eclgrid->getPinchThresholdThickness();
-                auto transMode = eclgrid->getPinchOption();
-                auto multzMode = eclgrid->getMultzOption();
-                PinchProcessor<Grid> pinch(minpv, thickness, transMode, multzMode);
 
-                std::vector<double> htrans_copy(htrans.size());
-                std::copy_n(htrans.data(), htrans.size(), htrans_copy.begin());
 
-                std::vector<int> actnum;
-                eclgrid->exportACTNUM(actnum);
-
-                auto transMult = eclState->getTransMult();
-                std::vector<double> multz(numCells, 0.0);
-                const int* global_cell = Opm::UgGridHelpers::globalCell(grid);
-
-                for (int i = 0; i < numCells; ++i) {
-                    multz[i] = transMult->getMultiplier(global_cell[i], Opm::FaceDir::ZPlus);
-                }
-
-                // Note the pore volume from eclState is used and not the pvol_ calculated above
-                const auto& porv = eclProps.getDoubleGridProperty("PORV").getData();
-                pinch.process(grid, htrans_copy, actnum, multz, porv, nnc_);
-            }           
+            // Use volume weighted arithmetic average of the NTG values for
+            // the cells effected by the current OPM cpgrid process algorithm
+            // for MINPV. Note that the change does not effect the pore volume calculations
+            // as the pore volume is currently defaulted to be comparable to ECLIPSE, but
+            // only the transmissibility calculations.
+            bool opmfil = eclgrid->getMinpvMode() == MinpvMode::ModeEnum::OpmFIL;
+            // opmfil is hardcoded to be true. i.e the volume weighting is always used
+            opmfil = true;
+            if (opmfil) {
+                minPvFillProps_(grid, eclState, ntg);
+            } else if (eclgrid->isPinchActive()) {
+                // opmfil is hardcoded to be true. i.e the pinch processor is never used
+                pinchProcess_(grid, *eclState, htrans, numCells);
```

Comment:
Getting the running of the `pinchProcessor` out of the body in this way was good!